End-of-2020 version

alex-mitrevski opened the pull request:
This PR adds all new features developed and bug fixes made during the year:
* Actions for human-robot object hand over (https://github.com/b-it-bots/mas_domestic_robotics/pull/227) and robot-human object hand over (https://github.com/b-it-bots/mas_domestic_robotics/pull/244)
* New scenarios:
    * A lab manager scenario (https://github.com/b-it-bots/mas_domestic_robotics/pull/233); a demonstration can be found at https://youtu.be/WqgsuoSAT8M
    * A tidy up scenario (https://github.com/b-it-bots/mas_domestic_robotics/pull/249), including a component for detecting (small) objects from point cloud measurements (https://github.com/b-it-bots/mas_domestic_robotics/pull/248); a demonstration is provided at https://youtu.be/Tnhf6nUeuDI
    * A demo scenario for receiving an object from a person and delivering it to to a kitchen (https://github.com/b-it-bots/mas_domestic_robotics/pull/246)
    * A utility scenario for learning the face of an unknown person (https://github.com/b-it-bots/mas_domestic_robotics/pull/235)
* Various changes and improvements to the existing components:
    * Object placing by detecting impact with the placing surface (https://github.com/b-it-bots/mas_domestic_robotics/pull/239)
    * Online DMP trajectory generation (https://github.com/b-it-bots/mas_domestic_robotics/pull/231, https://github.com/b-it-bots/mas_domestic_robotics/pull/240); this is required for https://github.com/b-it-bots/mas_domestic_robotics/pull/232
    * Integration of the `View` messages added in `mas_perception_msgs` (https://github.com/b-it-bots/mas_domestic_robotics/pull/234)
    * Rewriting of the action launchers so that they accept arguments (https://github.com/b-it-bots/mas_domestic_robotics/pull/242)
    * Replacement of the `ssd_keras` detector by a PyTorch-based detector (https://github.com/b-it-bots/mas_domestic_robotics/pull/245, https://github.com/b-it-bots/mas_domestic_robotics/pull/247)
* A new rosinstall file for setting up a development version of the repository and its dependencies (https://github.com/b-it-bots/mas_domestic_robotics/pull/237)
* `.pylintrc` and `.bandit` files for suppressing certain Codacy warnings (such as warnings about `random.choice` being unsafe or about class methods (base or overriden) that can be functions), which don't really fit our context (https://github.com/b-it-bots/mas_domestic_robotics/pull/251)
* Various documentation updates (https://github.com/b-it-bots/mas_domestic_robotics/pull/236, https://github.com/b-it-bots/mas_domestic_robotics/pull/238, https://github.com/b-it-bots/mas_domestic_robotics/pull/243)

**Edit 06.01.2021**: Should not be merged before https://github.com/b-it-bots/mas_domestic_robotics/pull/252

alex-mitrevski committed:
[handle_open_action] Modified the action to include handle detection

alex-mitrevski committed:
[handle_open_action/action_states] Error fix (undefined variable)

alex-mitrevski committed:
[handle_open_action] Error fix in wait_for_result call

alex-mitrevski committed:
[handle_open_action] Fixed a pose transform call with a non-existing object

af-a committed:
[move_arm_action] Add online DMP trajectory generation

af-a committed:
[hand_over_action] Update README with more detailed description

af-a committed:
[move_arm_action] Fix minor errors (tested on HSR)

af-a committed:
[move_arm_action] Add missing function docstring

af-a committed:
[move_arm_action] Correct references to np array variables in docstrings

sthoduka committed:
[hand_over_action] move arm to neutral position in init state (otherwise the execution of the DMP is weird); fix readme

DharminB committed:
[scenarios] Added skeleton for mdr_demo_lab_manager

DharminB committed:
[mdr_demo_lab_manager] Added states for lab manager (first draft)

af-a committed:
Add debug flag for calls to say()

af-a committed:
Add simple say sentence HRI behaviour

af-a committed:
Remove server launch files; add find people client launch file

af-a committed:
[demo_lab_manager] Modify monitor door state and add verify person state

alex-mitrevski committed:
[demo_lab_manager] Added stub verify_person implementation

alex-mitrevski committed:
[demo_lab_manager] Added a setup.py and fixed the structure of the python package

alex-mitrevski committed:
[demo_lab_manager] Forgot to set up the Python package in the CMakeLists

af-a committed:
[hri_behaviours] Fix incorrect scenario name

af-a committed:
[demo_lab_manager] Fix incorrect state name

alex-mitrevski committed:
[hri_behaviours] Added a state for asking for a name

alex-mitrevski committed:
[demo_lab_manager] Verify person can now send the robot to a random location

This is added just for testing purposes

alex-mitrevski committed:
[navigation_behaviours/move_base] Using destination_locations from userdata if not provided as an arg

alex-mitrevski committed:
[demo_lab_manager] Fixed missing userdata key declaration

alex-mitrevski committed:
[hri_behaviours/ask_name] Added missing userdata key declaration

alex-mitrevski committed:
[find_people_action/client] Fixed check for undetected people

Sushant-Chavan committed:
Replace speech state with the google forms based user data query

mas-hsr committed:
[demo_lab_manager/scenario_states/verify_person] Adding the occupancy checking implemented by Dharmin yesterday

af-a committed:
[find_people_action] Modify to integrate face image extraction and adding to Person msg

samuelpg committed:
[verify_person] Placeholder code for searching in known faces

Sushant-Chavan committed:
[get_user_data] Ask the person to scan the QR code printed in the wall

alex-mitrevski committed:
[find_people/action_states] The 'views' fields of the Person and Face msgs are now used

alex-mitrevski committed:
[find_my_mates/describe_person] Used the 'views' field of the Person msg

alex-mitrevski committed:
[demo_lab_manager/verify_person] The 'views' field of the Face msg is now used

alex-mitrevski committed:
[find_people/action_states] Removed unused imports; fixed a relative import

alex-mitrevski committed:
[demo_lab_manager/verify_person] Converted the embeddings to np arrays before use

alex-mitrevski committed:
[find_people] Face embeddings can now be also computed

Provided that a path to an embedding model is specified

alex-mitrevski committed:
[find_people] Fixed wrong import

samuelpg committed:
[verify_person] free up a space

alex-mitrevski committed:
[find_people/action_states] Fixed the use of the embedding model

alex-mitrevski committed:
[demo_lab_manager/verify_person] Added missing imports

alex-mitrevski committed:
[demo_lab_manager/verify_person] Refactored to use the 'recognise_person' fn from the KB interface

The function was added with
https://github.com/b-it-bots/mas_knowledge_base/pull/39

alex-mitrevski committed:
[scenarios] Added a metapackage mdr_utility_tasks for, well, small utility scenarios

alex-mitrevski committed:
[utility_tasks] Added a scenario for collecting images of an unknown person

alex-mitrevski committed:
[util_register_person] Added a missing dependency

alex-mitrevski committed:
[util_register_person/sm] Fixed YAML error (unescaped single quote)

alex-mitrevski committed:
[find_people/action_states] Fixed bug in which detected people weren't added to the list if there was no face image

alex-mitrevski committed:
[util_register_person/store_person] Removing the person msg on failure as well

alex-mitrevski committed:
[util_register_person/register_person.launch] Included the rosplan and find people action launchers

alex-mitrevski committed:
[util_register_person/sm] Corrected instruction sentence

To use the SaySentence state correctly, I am now passing just one
sentence to the state instead of two

alex-mitrevski committed:
[demo_lab_manager/get_user_data] __init__: input_key -> input_keys

samuelpg committed:
[lab_manager] added debig message

samuelpg committed:
[lab_manager/verify_person] added sleep after instruction

samuelpg committed:
[lab_manager/verify_person] added list of words that sound similar to 'bye'

samuelpg committed:
[lab_manager/verify_person] extended the sleep before listening for goodbye

samuelpg committed:
[lab_manager/verify_person] modified the state machine and checked for logged users in verify person

samuelpg committed:
[lab_manager/verify_person] fixed wrong indentation

alex-mitrevski committed:
[find_people] Fixed import for draw_labeled_boxes (from mas_perception_libs)

alex-mitrevski committed:
[environments/brsu-c069] Added poses for the lab manager scenario

alex-mitrevski committed:
[find_people/action_states] Removed grayscale img conversion before calculating a face embedding

alex-mitrevski committed:
[demo_lab_manager/verify_person] Assigned sitting spot also in the case of a known person entering the lab

samuelpg committed:
[lab_manager/verify_person] fixed no salutation bug

alex-mitrevski committed:
[demo_lab_manager] Added a state for sending a disinfection cmd

alex-mitrevski committed:
[demo_lab_manager] Recover in case a person is detected, but their face is not seen

alex-mitrevski committed:
[demo_lab_manager/perform_disinfection] Added instructions for disinfection and a missing transition

alex-mitrevski committed:
[demo_lab_manager/config] Increased duration of disinfection procedure

alex-mitrevski committed:
[demo_lab_manager/config] Added more instructions states

alex-mitrevski committed:
[hri_behaviours/say_sentence] Added an optional sleep argument for long sentences

alex-mitrevski committed:
[demo_lab_manager] Adeed sleep in the last instruction state; modified disinfection instruction sentence for clarity

alex-mitrevski committed:
[demo_lab_manager/launch] Included the ROSPlan launcher

alex-mitrevski committed:
[demo_lab_manager/verify_person] Update the KB when a person is assigned a spot

alex-mitrevski committed:
[demo_lab_manager/verify_person] Perform check for assigned spot only if a person is recognised

alex-mitrevski committed:
[perception_behaviours/find_people] Only say people aren't found in debug mode

alex-mitrevski committed:
[nav_behaviours/move_base] Added a debug mode for saying debugging messages

alex-mitrevski committed:
[demo_lab_manager/get_user_data] Fixed imports and KB updates

alex-mitrevski committed:
Resolved various Codacy issues

alex-mitrevski committed:
[demo_lab_manager/launch] Included the listen action launcher

Sushant-Chavan committed:
Add additional notes for building the packages on the robot

alex-mitrevski committed:
[demo_lab_manager/verify_person] Removing the person_0 msg from the KB before repeating the monitoring operation

person_0 is the KB ID of the person that we verify; if the message is
not removed and there are multiple messages with the same ID, the oldest
message will always be returned during retrieval, which is undesired

Sushant-Chavan committed:
Add a new rosinstall file with devel versions for repositories

Sushant-Chavan committed:
Use the master branch of b-it-bots fork for ROSPlan

Sushant-Chavan committed:
Update README instructions

Sushant-Chavan committed:
Remove redundant state in mdr_demo_simple_pick_and_place and cleanup

Sushant-Chavan committed:
Remove unused outcomes in demo scenarios

alex-mitrevski committed:
[gripper_controller] Added fns for impact detection along z (useful for placing)

I want to do impact detection as change detection over force measurements,
so I've added two functions:
* one for initialising the measurements
* one for checking if the difference between the initial measurements
  and the current measurements is above a threshold

alex-mitrevski committed:
[place_action] Added downward arm movement for placing by detecting surface impact

alex-mitrevski committed:
Merge branch 'devel' into feature/hand-over-action

alex-mitrevski committed:
Merge branch 'devel' of github.com:b-it-bots/mas_domestic_robotics into features/surface-impact-while-placing

alex-mitrevski committed:
[move_arm_action/dmp] Removed unused fn (generate_trajectory)

alex-mitrevski committed:
[move_arm_action/dmp] Fixed a transform bug in the DMP executor

The problem here was that the goal was given in the base link frame and
the initial position of the end effector was also taken wrt base link,
but during the execution, the poses were transformed to the odometry
frame. Both the initial pose and the goal pose are now transformed to
the odometry frame before execution starts.

af-a committed:
Add point cloud filtering to improve person pose estimation

af-a committed:
Utilize person pose for pre-approach (in action and scenario)

af-a committed:
[demo_hand_over] Add transition to detect_people if identify_posture fails

alex-mitrevski committed:
[hand_over_action] Fixed a small bug due to which the arm wasn't moving to the hand over pose

In particular, the timestamp of the hand over pose was set to the
current ROS time, so the pose could not be transformed during DMP
execution

af-a committed:
[hand_over_action] Add force sensor topic

af-a committed:
[hand_over_action] Update action client script

af-a committed:
[hand_over_action] Modify release detection strategy to use force readings in z-axis

af-a committed:
[hand_over_action] Remove hsr-specific topics

af-a committed:
[hand_over_action] Remove specification of post message timestamp

af-a committed:
[hand_over_action] Remove unnecessary comments and add move_base goal cancellation after motion duration

af-a committed:
Merge branch 'feature/hand-over-action' of github.com:b-it-bots/mas_domestic_robotics into feature/hand-over-action

alex-mitrevski committed:
Merge branch 'devel' of github.com:b-it-bots/mas_domestic_robotics into feature/hand-over-action

alex-mitrevski committed:
[hand_over_action/action_states] Resolved some Codacy complaints

alex-mitrevski committed:
[find_people/src/find_people] Removed unnecessary line

alex-mitrevski committed:
Merge pull request #227 from b-it-bots/feature/hand-over-action

Object Hand Over Action

alex-mitrevski committed:
[manipulation_actions] Use launch file args in the action server launchers

Related to https://github.com/b-it-bots/mas_domestic_robotics/issues/241

alex-mitrevski committed:
[navigation_actions] Use launch file args in the action server launchers

Related to https://github.com/b-it-bots/mas_domestic_robotics/issues/241

alex-mitrevski committed:
[perception_actions] Use launch file args in the action server launchers

Related to https://github.com/b-it-bots/mas_domestic_robotics/issues/241

alex-mitrevski committed:
[listen_action] Use launch file args in the action server launcher

Related to https://github.com/b-it-bots/mas_domestic_robotics/issues/241

alex-mitrevski committed:
[actions] Use launch file args in the action client launchers

Related to https://github.com/b-it-bots/mas_domestic_robotics/issues/241

alex-mitrevski committed:
[place_action] Added a new goal field to the action definition

The field is a Boolean "release_on_impact", which specifies whether the
object should be released by detecting impact with the surface or by
simply releasing the object (the latter is used for throwing an object)

alex-mitrevski committed:
[place_action/action_states] The release_on_impact goal field is now used by the action

alex-mitrevski committed:
[place_action/clients] Setting 'release_on_impact' in the place and throw clients

The field is set to true and false respectively (we want to detect
surface impact while placing, but we don't want to wait for impact while
throwing an object)

alex-mitrevski committed:
Merge branch 'devel' into features/surface-impact-while-placing

alex-mitrevski committed:
Merge pull request #239 from b-it-bots/features/surface-impact-while-placing

Impact-based object placing

sthoduka committed:
[hand_over_action] fix use of arguments in action client test and README

alex-mitrevski committed:
[manipulation_actions] Added an action for receiving an object from a person/another agent

alex-mitrevski committed:
[demos] Added context-aware object reception demo

alex-mitrevski committed:
[find_people] Replaced the SSD keras detector with a torch Faster R-CNN detector

alex-mitrevski committed:
[find_people/action_states] Fixed scope of newly added parameters

alex-mitrevski committed:
[find_people.launch] coco_classes.yaml -> coco_classes.yml

This is the name of the file in mas_perception_libs

alex-mitrevski committed:
[find_people] Fixed incorrectly passed argument to mas_perception_libs util fn

alex-mitrevski committed:
[find_people] Added some debugging msgs during execution

alex-mitrevski committed:
[find_people/client_test] Fixed a runtime error; added img publisher for debugging purposes

alex-mitrevski committed:
[find_people] Fixed a bug in the pose extraction call

alex-mitrevski committed:
[find_people/package.xml] Added Python package dependencies

alex-mitrevski committed:
[receive_object/action_states] Changed the sign of the mean for CUSUM

alex-mitrevski committed:
[robocup_tasks] Added a skeleton package for a METRICS drink transportation task

alex-mitrevski committed:
[find_people/client] Added optional person recognition and msg removal from the KB

alex-mitrevski committed:
[perception_behaviours/find_people] A person recognition arg is now passed to the action client

alex-mitrevski committed:
[hri_behaviours] Added an object reception behaviour

This behaviour
(i) estimates the posture of a person and
(ii) calls the 'receive_object' action for receiving an object

alex-mitrevski committed:
[receive_and_transport_drink/config] Added an initial SM for the task

alex-mitrevski committed:
[gripper_controller/base] Added a 'get_opening_angle' function

alex-mitrevski committed:
[gripper_controller/base] Added a 'get_gripper_pose' function

af-a committed:
[hri_behaviours/receive_object] Add object size estimation

af-a committed:
[hri_behaviours/receive_object] Set position of grasped object to position of gripper

alex-mitrevski committed:
[receive_and_transport_drink] Added a missing transition to the SM config

alex-mitrevski committed:
[hri_behaviours/receive_object] Fixed incorrect default action server value

af-a committed:
[hri_behaviours/receive_object] Fix incorrect action parameter name

af-a committed:
Fixed various errors related to receive ans transport drink task

af-a committed:
[mdr_hri_behaviours/receive_object] Include object category for object to transport

alex-mitrevski committed:
[perceive_plane/action_states] Only create a recognition proxy if this feature is enabled

alex-mitrevski committed:
[perceive_plane] Added a config dir and an object detector config file

alex-mitrevski committed:
[perceive_plane] Launch a plane detector within the action launcher

The plane detector uses a PyTorch-based object detector (config
parameters specified in mdr_perceive_plane_action/config)

alex-mitrevski committed:
[mas-domestic-devel.rosinstall] Added demonstrated_trajectory_recorder to the list of dependencies

Sushant-Chavan committed:
Add a new scenario package and launch file for the wrs-2020-challenge

alex-mitrevski committed:
[scenarios] mdr_wrs_2020_challenge -> mdr_wrs_tidy_up

alex-mitrevski committed:
[wrs_tidy_up] Added SM draft (without object delivery task)

alex-mitrevski committed:
[wrs_tidy_up/config] Added a YAML file mapping from object classes to storage locations

For this particular case, the file acts as a "pseudo-ontology" (this is
essentially a model of the defaultStoringLocation relation in our
ontology).

We don't use the ontology for the scenario since the model
requires a reasoner, which is only supported by owlready2, which in turn
doesn't support Python 2 anymore (and we are still stuck with Python 2).

alex-mitrevski committed:
[environments] Added an environment for WRS2020

The navigation goals file currently only includes poses in the first
room

alex-mitrevski committed:
[move_arm_action] Included whole body moveit group

This is an optional MoveIt! group that may be required for moving a
robot to arbitrary 3D poses.

The group is only initialised and used if a name for the group is passed;
otherwise, only the arm group will be used during execution.

af-a committed:
[mdr_manipulation_msgs] Add UpdatePlanningScene service

alex-mitrevski committed:
[move_arm] Added a server for updating the planning scene

The service is of the newly added type mdr_manipulation_msgs/UpdatePlanningScene

alex-mitrevski committed:
[wrs_tidy_up] Added mdr_manipulation_msgs to the list of dependencies

af-a committed:
Split cleanup_object state, and add missing module names

alex-mitrevski committed:
[tidy_up/config] Small updates to the SM

af-a committed:
[wrs_tidy_up] Add SelectObjectForPicking state

alex-mitrevski committed:
[wrs_tidy_up/config] Various fixes to the SM

alex-mitrevski committed:
[wrs_tidy_up] Added state for scenario initialisation

The initialisation of the MoveIt! planning scene is missing here;
this will be added with a bit later on.

alex-mitrevski committed:
[wrs_tidy_up] Added a state that takes care of grasping an object

alex-mitrevski committed:
[wrs_tidy_up/select_object_for_picking] Declared missing input keys

alex-mitrevski committed:
[wrs_tidy_up/pickup_object] Set a 'grasped_object' output and added the grasped object to the KB

These are both necessary so that the mdr_manipulation_behaviours.throw_object_in state can be used

alex-mitrevski committed:
[wrs_tidy_up/config] Added a config file for initialising the moveit planning scene with static environment objects

All tables, the two bins, and all walls have been added to the scene;
the bookshelf and the drawers have not been added in this version

alex-mitrevski committed:
[wrs_tidy_up/initialise_scenario] The moveit planning scene is now initialised

alex-mitrevski committed:
[wrs_tidy_up/config/storage_locations] box -> bin; added bin as default location for unknown objects

alex-mitrevski committed:
[wrs_tidy_up/initialise_scenario] The KB is now initialised as well

alex-mitrevski committed:
environments/wrs20-arena] Added poses for the second room

alex-mitrevski committed:
[wrs_tidy_up/pickup_object] Added a grasping height offset

This avoids collisions of the gripper with the surface on which the
object is

alex-mitrevski committed:
[wrs_tidy_up] Added new states; modify existing ones to prevent interaction with MongoDB

alex-mitrevski committed:
[wrs_tidy_up/config] Commented out states for T1; updated state modules with new states

alex-mitrevski committed:
[*.rosinstall] Added mas_tools to the dependency list

alex-mitrevski committed:
[wrs_tidy_up/states/move_base] Fixed undefined var error

alex-mitrevski committed:
[wrs_tidy_up] Added state for selecting a scanning pose (Task 1)

alex-mitrevski committed:
[wrs_tidy_up/states/move_base] Don't convert potentially None argument to list

alex-mitrevski committed:
[wrs_tidy_up/states/init_scenario] Convert args to dict before adding them to the userdata

They otherwise had the type "OrderedDict", which is then stored as
smach.user_data.Const

alex-mitrevski committed:
[wrs_tidy_up/config] The basic workflow for task 1 is now there

alex-mitrevski committed:
[wrs_tidy_up/config] Removed one remaining reference to mdr_behaviours (use move_base from scenario instead)

alex-mitrevski committed:
[wrs_tidy_up] Added state for selecting a storage location

alex-mitrevski committed:
[wrs_tidy_up/states/init_scenario] Use mas_tools fn for reading YAML file

alex-mitrevski committed:
[wrs_tidy_up/states/pickup_object] Added sideways grasp for tall objects

alex-mitrevski committed:
[.travis.yml] Added pip upgrade command

alex-mitrevski committed:
[wrs_tidy_up/states/select_storage_location] Removed unnecessary debugging msgs

af-a committed:
[wrs_tidy_up/states/pickup] Added grasp strategy selection

Sushant-Chavan committed:
Add dynamic reconfigure clients to change the object detection setup

Sushant-Chavan committed:
Add a config file to store object detection params

alex-mitrevski committed:
[wrs_tidy_up/utils] Fixed fn argument use

alex-mitrevski committed:
[mdr_wrs_tidy_up/utils] Fixed path to detection params config file

alex-mitrevski committed:
[wrs_tidy_up/config] Added some missing objects to the scene map

In particular, I've added:
* the trays on the placing table
* the containers on the placing table (even though they're not required for
  the scenario)
* the tables have horizontal cylindrical bars below the table plane, so
  I've added a collision box for those

alex-mitrevski committed:
[wrs_tidy_up/states/init] Skipping objects with support in the name; added more debugging msgs

alex-mitrevski committed:
[wrs_tidy_up/states/move_base] Fixed bug with source of locations

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Set floor location to 'cleared' if no objects found

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Set grasping pose to top of object for top-down grasp

This is to prevent pushing down on the object when grasping from the top

alex-mitrevski committed:
[wrs_tidy_up/config/object_detection_params] Reduced the min-z cutoff for floor objects

In particular, the value is now 0.02 (the previous value was 0.01). With
the old value, small objects (such as a pen and a fork) were missed by
the detection component.

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Sleep for 0.5s after clearing the object cache

alex-mitrevski committed:
[wrs_tidy_up/states/select_object] Ignore z when looking for closest object

alex-mitrevski committed:
[wrs_tidy_up/config/sm] Removed bin_b as a possible placing location

bin_a seems to be easier for placing, so it's better to keep only that
one

alex-mitrevski committed:
[wrd_tidy_up/states/find_objects] Filter objects based on height; for now, use same detection method for all locations

The height-based filtering is to avoid Gazebo issues with grasping the
pitcher, which seems to stick to the gripper

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Increases grasp offset to 10cm; set the z-orientation correctly during a top-down grasp

alex-mitrevski committed:
[environments/wrs20-arena] Moved one inspection pose a bit further from the table

Otherwise there were cases in which objects were lying on the pose and
move_base couldn't find a plan

alex-mitrevski committed:
[move_base_action] Added simple recovery behaviour if case goal pose is unreachable

The recovery generates a new goal pose (by adding Gaussian noise to the
original xy-position) and retries the navigation

alex-mitrevski committed:
[environment/wrs20-arena] Updated the scanning poses

The same scanning position is now used, but with three different
orientations; this covers the complete table area and reduces the
possibility for navigation failures due to objects occupying the goal
pose

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Added filtering based on distance to the closest table leg

The path planner generally fails when objects are too
close to the table legs; as a workaround, I decided to filter out such
objects

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Re-introduced a height offset for top-down grasps

alex-mitrevski committed:
[wrs_tidy_up/states/release] Set the release pose to the center of the release target

This seems more reliable than randomly sampling a position on the target
area, which might be very close to the edge and thus lead to a failure

alex-mitrevski committed:
[wrs_tidy_up/config/sm] Increased grasping timeout to 45s

In rare cases, path and motion planning takes quite a while; 45s is a
generous timeout that allows such cases to be accepted as well

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Added filtering for objects under the tables

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Set cleared locations depending on object location

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Filter objects based on distance to table legs only when cleaning the floor

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Added goal pose pub for debugging purposes

alex-mitrevski committed:
[wrs_tidy_up] Added timeout of 15min for task 1

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Removed the grasping offset for top-down grasps

This seemed to cause problems for large objects

alex-mitrevski committed:
[wrs_tidy_up/config/sm] Retry pickup only once

If things start going wrong (e.g. an object slips continuously),
the robot gets stuck in the pickup state for too long without making any
progress in the task

alex-mitrevski committed:
[wrs_tidy_up/states/select_object_for_picking] Added debugging msg for position and size of selected object

Sushant-Chavan committed:
[wrs_tidy_up/states/find_objects]  Filter large and flat objects on ground

alex-mitrevski committed:
[wrs_tidy_up/config/sm] Removed grasp retries

For increased reliability, it seems better to re-perceive the scene
since the previous grasping may have caused changes

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Static objects are now added to the planning scene before grasping (and removed afterwards)

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] Reintroduced a small grasping offset for top-down grasps

Otherwise, planning seems to fail quite often for objects on tables
(due to collisions with the table plane)

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Added object detection timeout; filtered objects between the two tables

alex-mitrevski committed:
[wrs_tidy_up/states/init] Fixed initialisation for task 2

The initialisation was crashing without checks on the arguments (as some
of them are not passed for the second task)

alex-mitrevski committed:
[wrs_tidy_up/config/sm] Increased pickup timeout and removed tries for obstacle picking state

alex-mitrevski committed:
[wrs_tidy_up/states/find_objects] Set location to cleared if object detection timeout reached

alex-mitrevski committed:
[wrs_tidy_up/config/planning_scene] Made the walls higher (0.6m -> 2m)

This is to prevent moveit from creating plans from outside a room

Sushant-Chavan committed:
[wrs_tidy_up/config/object detection] Tweaked the z_min point cloud cuttoff to detect objects on tables and shelves

alex-mitrevski committed:
[wrs_tidy_up/config/planning_scene] Made tables narrower

alex-mitrevski committed:
[wrs_tidy_up/states/pickup] After all tests, a height offset of 5cm seems the most reliable in general

Sushant-Chavan committed:
Copy over the cloud object detection code from mas_perception_libs and remove dependency on mas_perception_libs

Sushant-Chavan committed:
Move individual classes to separate header/cpp files

Sushant-Chavan committed:
Cleanup and fixes

Sushant-Chavan committed:
Remove comments from CMakeLists.txt and package.xml

Sushant-Chavan committed:
Fix errors in creating the object messages for each cluster

Sushant-Chavan committed:
Use the launch file of the occupancy_checker

alex-mitrevski committed:
[find_people/package.xml] Removed pip packages from the dependency list

rosdep is apparently not able to find any of them

alex-mitrevski committed:
[cloud_object_detection] Added tf2_geometry_msgs to the list of dependencies

alex-mitrevski committed:
[cloud_object_detection/CMakeLists] Added catkin dependency on dynamic_reconfigure

alex-mitrevski committed:
[manipulation_msgs] Added actionlib to the list of dependencies

The build is failing otherwise (even though the package doesn't have a
**direct** dependency on it)

alex-mitrevski committed:
[hri_behaviours/receive_object] Bug fixes

* Fixed incorrect object category setting
* Removed unreachable outcome

alex-mitrevski committed:
Resolved various Coday issues (mostly unused imports)

Sushant-Chavan committed:
[Codacy-Fix] Remove unused imports

Sushant-Chavan committed:
[Codacy] Add configuration files to disable some tests in pylint and bandit

alex-mitrevski committed:
Resolved further Codacy issues (also older ones)

Sushant-Chavan committed:
Merge branch 'devel' into devel

Sushant-Chavan committed:
[Codacy] add comments to the bandit config file

alex-mitrevski committed:
Merge pull request #251 from Sushant-Chavan/devel

Fix Codacy errors and add configuration files to skip some pylint and bandit tests

alex-mitrevski committed:
[README] Added a link to mas_tutorials

alex-mitrevski committed:
[.bandit] Passed warnings as a list and removed [bandit] tag

Sushant-Chavan committed:
[.bandit] Add comments about the skipped tests

minhnh commented on mdr_perception/mdr_cloud_object_detection/ros/include/mdr_cloud_object_detection/Aliases.h:
```diff
@@ -0,0 +1,18 @@
+#include <pcl/point_types.h>
+#include <pcl/point_cloud.h>
+
+typedef pcl::PointXYZRGB PointT;
```
@Sushant-Chavan isn't this definitions already in `mas_perception_libs`?

Sushant-Chavan replied:
Yes. These are basically copied over from `mas_perception_libs`. I could not use the headers defined in `mas_perception_libs` in the new `mdr_cloud_object_detection` package since these headers are not available at build time since the mas_perception_libs is not "installed". 

Please correct me if I am wrong about this. But as far as I understand, the headers `Color.h` and `Aliases.h` will only be available for reference from the `mdr_cloud_object_detection` if the mas_perception_libs is "installed". Right?

alex-mitrevski replied:
@Sushant-Chavan I think if `mas_perception_libs` is specified as a dependency in the package.xml. it will be built before `mdr_cloud_object_detection` and the headers will be found even though the package is not installed. So I think this shouldn't be a problem.

Sushant-Chavan replied:
You are right. I actually tried it but then was using Uppercase first leteres for the headers which is why it was failing. That led me to believe that we need to explicity export the headers explicitly. It works now :)

minhnh commented on mdr_perception/mdr_cloud_object_detection/ros/include/mdr_cloud_object_detection/Color.h:
```diff
@@ -0,0 +1,108 @@
+/*!
```
@Sushant-Chavan same comment as above: is this redundant with what's in `mas_perception_libs`?

minhnh commented on mdr_perception/mdr_cloud_object_detection/ros/include/mdr_cloud_object_detection/PassThroughVoxelFilterParams.h:
```diff
@@ -0,0 +1,32 @@
+/*!
```
@Sushant-Chavan is this redundant with [`CloudFilter` class in `mas_perception_libs`](https://github.com/b-it-bots/mas_perception_libs/blob/0a9cf791c9b744a3ce1c23f26755bf2fbcd366db/common/include/mas_perception_libs/point_cloud_utils.h#L55)?

Sushant-Chavan replied:
This class simply adds a PassThrough filter for the Z-dimension in addition to what the `CloudFilter` class does. But I just found out that there is a [pcl::CropBox](https://pointclouds.org/documentation/classpcl_1_1_crop_box_3_01pcl_1_1_p_c_l_point_cloud2_01_4.html) filter which might be the best choice here. An example usage is shown [here](https://stackoverflow.com/a/45793962). I will change this class to replace all the `pcl::PassThrough` by a single `pcl::CropBox` filter.

minhnh replied:
So like mentioned in the comment below, you can crop z-axis using the voxel grid, it works basically the same, which was why I didn't have a passthrough filter for the z-axis before. You can indeed use `CropBox`, if you don't need to down-sample the point cloud.

Sushant-Chavan replied:
I do need to downsample the cloud. But I think using cropbox (even if we are just filtering based on just x and y dimensions) can be faster as we need to iterate over the cloud once rather than doing it twice with two PassThrough filters.

minhnh commented on mdr_perception/mdr_cloud_object_detection/ros/include/mdr_cloud_object_detection/PassThroughVoxelFilterParams.h:
```diff
@@ -0,0 +1,32 @@
+        float mPassThroughLimitMaxX = 0.0f;
+        float mPassThroughLimitMinY = 0.0f;
+        float mPassThroughLimitMaxY = 0.0f;
+        float mPassThroughLimitMinZ = 0.0f;
```
This is redundant with the z-axis filter of the `VoxelGrid` I think. It maybe more explicit, but do remember that every single point will be compared to the pass through limits one extra time.

minhnh commented on mdr_planning/mdr_scenarios/mdr_demos/mdr_demo_lab_manager/ros/src/mdr_demo_lab_manager/scenario_states/get_user_data.py:
```diff
@@ -0,0 +1,91 @@
+        rospy.logerr("[get_user_data] Query timed out!")
+
+        if self.retry_count == self.number_of_retries:
+            self.say("Sorry, I could get our name. I shall proceed without it.")
```
"I could not get your name?"

Sushant-Chavan committed:
Fix review findings

alex-mitrevski committed:
[move_arm_action/dmp] Fixed the online execution

The existing implementation was unrolling the DMP (i.e. calling the
dmp.step function) at every step of the control loop, without checking
if the intermediate goal has been reached. For this reason, the robot
was very quickly just starting to move directly towards the goal, thus
not following the trajectory at all.

In the fixed version, a DMP step is only taken if the currently pursued
goal has been reached, which means that the trajectory is actually
followed.

alex-mitrevski committed:
[pickup/place_action] Fixed test for empty goal orientation

Both None and empty list have to be covered
